wallet/file-format.md

```diff
@@ -107,9 +107,10 @@ In order to decrypt the private key, following steps are followed:
 
 1. User inputs a password.
 2. Password is converted to bytes (utf-8 encoding).
-3. Symmetric key is derived using PBKDF2 with this password (bytes) and the salt (bytes) as input.
-4. This derived symmetric key is then used as AES-GCM Key along with the nonce (IV) to decrypt the ciphered private key.
-5. The authentication tag at the end of ciphered data checks if the derived symmetric key used in point 4. is right.
+3. Salt is decoded from unpadded base64 to bytes.
```

Comment:
In the Rust library, the `Salt` struct is constructed from the base64-encoded (and unpadded) salt
However in Go, the Salt is used as a raw array of bytes.

In general, the salt is just an array of bytes, and used in the algorithms this way, the rest is just differences in the API of the libraries. So in our case, we use the base64-encoded value as-is in Rust, and decode it in Go, it depends.
(Hope that answers the question ?)

Reply:
> But from my discussion with @Eitu33 I understood that we must use the base64 representation of the salt as a byte array instead.

That was the case yes but I have not followed the evolution of this matter since our last discussion, I can't confirm that this is still the current expected behaviour.

Reply:
If the content of the value is decoded before being used by the cryptography library, then why change the standard? Why don't you encode it in base64 and then decode it on the rust code side ?